### ISSUE TITLE
New data_processing_files field in DataProcessing

### ISSUE THREAD

**Issue by bcorrie** [airr-community/airr-standards#276]
The DataProcessing object captures the processing that has been performed on some sequencing data. From a file perspective, we track the files from the output of a sequencing run. From a data provenance perspective, it seems helpful to provide a field to track the files produced by a DataProcessing step. We certainly track this, and from a data provenance perspective (for a data curator) it is very helpful to be able to link a file to a set of rearrangement data. 

I would suggest something like 

```
        data_processing_file:
            type: string
            description: File name for data produced by a data processing step.
            example: MS10R-NMonson-C7JR9_S1_R1_001.igblast.airr.tsv
            x-airr:
                miairr: false
                required: false
                nullable: true
                set: 5
                subset: process (computational)
                name: Processed data file name
```
This is related to #275 where I am suggesting that it would be useful to have something like:

"data_processing_file":"TW05B_T_naive_CD8.igblast.airr.tsv"

as part of the data_processing record for the Florian data.

**Comment by schristley** [comment]
> files produced by a DataProcessing step

what's a step? You said "files" as in one or more ;-D

I probably understand the intent though. I have something similar in VDJServer (a custom field) to record the actual file name that holds the data before it's loaded into the data repository. It might have usefulness to a broader audience somewhere down the line, but right now it is pretty specific to the VDJServer process.

> it would be useful to have

Why would it be useful? Is it useful to anybody other than the original data curator?

If we can come up with clear semantics and usefulness to general users, then I think it belongs in the schema, otherwise it might be better as custom field.

**Comment by bcorrie** [comment]
> > files produced by a DataProcessing step
> 
> what's a step? You said "files" as in one or more ;-D

I was kind of thinking forward to capturing something clone processing as a data processing object as well. For a given repertoire, one would have annotation data processing step, and record the files for that data processing output (e.g. TW05B_T_naive_CD8.igblast.airr.tsv). One could also have a clonal analysis data processing step, store all of the relevant information in that data processing object, including output that data processing step (e.g. TW05B_T_naive_CD8.changeo.out).

> Why would it be useful? Is it useful to anybody other than the original data curator?

Possibly not, but if you find it useful and we find it useful, that seems like it is at least worth considering. I also think it is as useful as recording the information that we record in the SequencingFiles object, which are also primarily useful to the data curator, no? So there is a precedence for this.

At the same time, I think it is important that we help data curators manage data provenance in the standard as well, as they are a primary user of the data standards.

**Comment by schristley** [comment]
fixed in #287
